### PR TITLE
Implement various TPM functions

### PR DESCRIPTION
**Pull request by puiterwijk**
This implements the following TPM2 functions:
- TPM2_Hmac
- TPM2_Shutdown
- TPM2_SelfTest
- TPM2_GetTestResult
- TPM2_PolicyRestart
- TPM2_StirRandom

### CODE REVIEW

**Comment by puiterwijk** [comment]
I will rebase this on top of #179 when that gets merged.

**Review by ionut-arm — APPROVED** [review]
It's really nice seeing Rusty interfaces on all `Context` methods 🤩 this codebase has improved so much in the past year!!

**Review by Superhepper — COMMENTED** [review]
Overall it looks good. One thing I would like to see perhaps are some code examples in the documentation.

File: tss-esapi/src/utils/mod.rs [hunk 1]
```diff
@@ -413,6 +451,65 @@ impl TpmsEccParmsBuilder {
...
+        self.hash_alg = hash_alg.into();
+        self
+    }
+
+    pub fn with_kdf(mut self, kdf: TPM2_ALG_ID) -> Self {
+        self.kdf = kdf;
+        self
+    }
+
+    pub fn build(self) -> Result<TPMS_KEYEDHASH_PARMS> {
```
[the remark this answers is not in the document]
Comment (puiterwijk) [reply tss-esapi/src/utils/mod.rs]: We now have `crate::structures::KeyedHashParms`.

**Review by Superhepper — APPROVED** [review]

**Pull request merged**